torchao/quantization/quantize_/workflows/int4/int4_plain_int32_tensor.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+from typing import List
+
+import torch
+
+from torchao.quantization.quant_primitives import (
+    MappingType,
+    _choose_qparams_affine,
+    _quantize_affine,
+)
+from torchao.utils import (
+    TorchAOBaseTensor,
+)
+
+__all__ = [
+    "Int4PlainInt32",
+]
+
+aten = torch.ops.aten
+
+
+class Int4PlainInt32(TorchAOBaseTensor):
+    """
+    int4 weight-only quantization on XPU with oneDNN as backend (groupwise quantization only)
+
+    Tensor Attributes:
+        qdata: packed int4 weigh, always viewed as a 2D (N, K/2) tensor
+        scale: (K/group_size, N), dtype is the same as the original Tensor dtype
+        zero_point: (K/group_size, N)
+
+    Non-Tensor Attributes:
+        block_size: the block size for quantization, representing the granularity, for groupwise quantization, will have block_size (1, group_size).
+                    we only support group_size = 32/64/128.
```

Comment:
add an assert in `from_hp`?

Reply:
Corrected. There is no block size limitation for XPU backend.